src/classes/BDI_AdditionalObjectService.cls

```diff
@@ -0,0 +1,367 @@
+
+    private String namespace = UTIL_Namespace.getNamespace();
+
+    /*******************************************************************************************************
+    * @description Constructor
+    * @param bdi The data import service instance that invokes us
+    */
+    public BDI_AdditionalObjectService(BDI_DataImportService bdi) {
+        this.bdi = bdi;
+        this.bdiFM = BDI_FieldMappingCustomMetadata.getInstance();
+    }
+
+
+    /*******************************************************************************************************
+    * @description Main method to call to begin the additional object import process.
+    */
+    public void importAdditionalObjects(){
+
+        this.generateTiersOfObjMappings();
+        this.createSetOfImportedRecordFieldNames(bdiFM.diomDevNameToDiomMap.values());
+
+        //Iterate through the map of tiers, starting with the first tier that actually has a predecessor (ie 2nd tier), and
+        //validate and create each object as appropriate.
+        for (Integer i = 1; i < tierToObjMappings.size(); i++) {
+
+            Map<String,BDI_ObjectWrap[]> objAPINameToObjWraps = new Map<String,BDI_ObjectWrap[]>();
+
+            Map<String,Data_Import_Object_Mapping__mdt> currentTier = tierToObjMappings.get(i);
+
+            for (Data_Import_Object_Mapping__mdt diom : currentTier.values()) {
+
+                BDI_ObjectMappingLogic objMappingLogic = getObjectMappingLogicClass(diom);
+
+                Map<Id,DataImport__c> validDIsForObjMapping = new Map<Id,DataImport__c>(bdi.listDI);
+
+                Data_Import_Object_Mapping__mdt predecessor = bdiFM.diomDevNameToDiomMap.get(diom.Predecessor__c);
+                Data_Import_Field_Mapping__mdt[] fieldMappings = bdiFM.diomDevNameToDifmsMap.get(diom.DeveloperName);
+
+                //Only proceed if at least one field mapping exists
+                if (fieldMappings != null && fieldMappings.size() > 0) {
+                    //Map<Id,DataImport__c> disForUpdate = new Map<Id,DataImport__c>();
+                    BDI_ObjectWrap[] objWrapsForProcessing = new BDI_ObjectWrap[]{};
+                    for (DataImport__c di: bdi.listDI) {
+
+                        // Skip over records that have already failed
+                        if (di.Status__c == BDI_DataImport_API.bdiFailed) {
+                            continue;
+                        }
+
+                        BDI_ObjectWrap objWrap = new BDI_ObjectWrap(di,diom,fieldMappings);
+
+                        //null out any existing value in the status field for this object mapping
+                        di.put(UTIL_Namespace.adjustNPSPNamespace(diom.Imported_Record_Status_Field_Name__c),'');
+                        String lcFieldName = UTIL_Namespace.adjustNPSPNamespace(diom.Imported_Record_Field_Name__c).toLowerCase();
+
+
+                        //If the imported record field for the object is not populated then create a new Sobject, otherwise update existing.
```

Comment:
This isn't a big thing, but these line wraps on the comments can make the code harder to read

Reply:
This line wrapping is done by github because of the insanely narrow width they have for viewing code.  Are you suggesting we format everything around this limit?

Reply:
I mean, when possible - it's not an absolute requirement but it helps. For comments especially, where we have a lot more flexibility around breaking lines. 

I've always assumed the narrow review window was actually intentional to enforce good practice around line breaks, but I might just be speculating?

Reply:
Just out of curiosity here, I checked with our style guide: "Try to keep lines under 120 characters to ensure better readability." There's good research that backs that up for readability as well. I think the github window is 120 characters?